### PR TITLE
fix: datamodel set_state

### PR DESCRIPTION
**Pull request by prmukherj**

### CODE REVIEW

File: src/ansys/fluent/core/services/datamodel_se.py [hunk 1]
```diff
@@ -886,9 +886,15 @@ def set_state(self, state: Optional[Any] = None, **kwargs) -> None:
...
         Raises
         ------
+        TypeError
+            If no arguments are provided to set_state.
         ReadOnlyObjectError
             If the object is read-only.
         """
+        if not state and not kwargs:
+            raise TypeError(
+                "set_state() requires either 'state' or 'kwargs' arguments."
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/services/datamodel_se.py:896]: Do we know the use case for kwargs?

Comment (prmukherj) [reply src/ansys/fluent/core/services/datamodel_se.py:896]: Yes @seanpearsonuk. Updated the test.
"meshing.GlobalSettings.set_state(EnableOversetMeshing=False)"

**Review by mkundu1 — APPROVED** [review]

File: src/ansys/fluent/core/services/datamodel_se.py (hunk 1 above)
Comment (seanpearsonuk) [reply src/ansys/fluent/core/services/datamodel_se.py:896]: Do flobject-based APIs support it, too?

Comment (mkundu1) [reply src/ansys/fluent/core/services/datamodel_se.py:896]: Yes, https://github.com/ansys/pyfluent/blob/main/tests/test_flobject.py#L443

**Review by seanpearsonuk — APPROVED** [review]

File: src/ansys/fluent/core/services/datamodel_se.py [hunk 2]
```diff
@@ -886,9 +886,15 @@ def set_state(self, state: Optional[Any] = None, **kwargs) -> None:
 
         Raises
         ------
+        TypeError
+            If no arguments are provided to set_state.
```
Comment (PipKat) [inline src/ansys/fluent/core/services/datamodel_se.py:890]:
```suggestion
            If no arguments are provided to the ``set_state()`` method.
```

File: src/ansys/fluent/core/services/datamodel_se.py (hunk 1 above)
Comment (PipKat) [inline src/ansys/fluent/core/services/datamodel_se.py:896]:
```suggestion
                "The `set_state()` method requires either 'state' or 'kwargs' arguments."
```

Comment (prmukherj) [reply src/ansys/fluent/core/services/datamodel_se.py:896]: Thank you @mkundu1 .

**Pull request merged**